### PR TITLE
Feat seo changes

### PR DESCRIPTION
**Pull request by moaiandin**
Changes to titles & SEO fixes to page structure and URLs

### CODE REVIEW

File: netlify/forms/request-demo.yaml [hunk 1]
```diff
@@ -11,6 +11,7 @@ hubspot:
 meta:
   favicon: /images/mark_light_bg.png
   title: Get a demo of our API Toolkit | Stoplight
+  description: Want to see how we could help your company and organisation with our APIs? Click here to request a demo and find out what we can do for you.
```
Comment (tbarn) [inline netlify/forms/request-demo.yaml]: "with our APIs" doesn't make sense. Possibly change it to "our API Toolkit?"

File: netlify/lists/blog.yaml [hunk 2]
```diff
@@ -21,6 +21,7 @@ actionBar:
 meta:
   favicon: /images/mark_light_bg.png
   title: Latest Stoplight Blog Posts | Stoplight API Corner
+  description: Find out all you need to know about the latest API related news & guides from the official Stoplight Blog, written by some of the best minds in the business.
```
Comment (tbarn) [inline netlify/lists/blog.yaml]: Could we tweak this one? 

Some suggestions: 
- Find out all you need to know about API design, testing, mocking, and documentation from the official Stoplight Blog
- Find out all you need to know about API development related news & guides from the official Stoplight blog

I want to make sure people know it isn't like API news like a ProgrammableWeb or something and more about development and building APIs

File: netlify/pages/about.yaml [hunk 3]
```diff
@@ -156,6 +156,7 @@ actionBar:
 meta:
   favicon: /images/mark_light_bg.png
   title: Learn About Us | Stoplight
+  description: Find out all you need to know about Stoplight and its team of talented and creative engineers; united in their passion for creating best in class API tooling.
```
Comment (tbarn) [inline netlify/pages/about.yaml]: replace "engineers" with a word that includes everyone on the team.

**Comment by moaiandin** [comment]
Changes made

**Pull request merged**